### ISSUE TITLE
[Bug]: 1.5.0 release could trigger code analyzers warnings/errors in consuming projects.

### ISSUE THREAD

**Issue by bjornen77** [CoreWCF/CoreWCF#1276]
### Duplicate ?

- [X] I have searched issues/discussions and did not find other issues/discussions reporting this bug.

### Product version

1.5.0

### Describe expected behavior

When upgrading to latest version code analyzers shall not emit any new warnings.

### Describe actual behavior

When upgrading to latest version of CoreWCF our code analyzers started to emit warnings about a .cs not following our analyzer rules. It seems like the following PR(https://github.com/CoreWCF/CoreWCF/pull/1235) introduces a new file(CoreWCFWebApplicationExtensions.cs) that will be part of the compilation. The issue is that this file might not comply with the analyzer settings you have in your project.

Is there some other solution for the original issue that won´t require files to be included in the nuget package? If not, maybe rename the file to "CoreWCFWebApplicationExtensions.g.cs" that disables analyzers for the file?

### Which binding

NetTcp

### security

None

### Which .NET version

.NET 7

### Which os platform

Windows

**Comment by bjornen77** [comment]
@g7ed6e

**Comment by mconnew** [comment]
A bit of background, we now have the extension method UseServiceModel for IHost and IApplicationBuilder. The reason being that you might not have an IApplicationBuilder if using the generic host. In .NET 6, there's the WebApplication class which implements both of these interfaces, which means the compiler can't work out which extension method to use and your code will fail to compile.  

The solution is to provide an extension method just for WebApplication as the compiler will choose that over casting to an interface. WebApplication isn't available to netstandard so we can't include that method in our implementation assembly without multi-targeting, which doubles the size of the package as it now carries two implementations.  

If this is causing you problems and you need to fix it before we release a patch (1.5.0 is done and built, just waiting for other external things to release), as it's being added in a .props file, you can remove the file from being compiled by adding this to your .csproj:

```xml
  <ItemGroup>
    <Compile Exclude="**/contentFiles/CoreWCFWebApplicationExtensions.cs" />
  </ItemGroup>
```

This removes the file from the compile item group so it no longer gets built. I you are calling UseServiceModel on WebApplication, you can cast it to IApplicationBuilder to fix this issue.
```c#
((IApplicationBuilder)app).UseServiceModel(builder => { /* ... */} );
```

**Comment by bjornen77** [comment]
@mconnew Thanks for the explanation and workaround!
